src/query/service/src/interpreters/interpreter_user_stage_create.rs

```diff
@@ -79,18 +82,26 @@ impl Interpreter for CreateUserStageInterpreter {
             )));
         };
 
-        if user_stage.stage_type != StageType::External {
-            let op = self.ctx.get_data_operator()?.operator();
-            op.create_dir(&user_stage.stage_prefix()).await?
-        }
-
         let mut user_stage = user_stage;
         user_stage.creator = Some(self.ctx.get_current_user()?.identity());
         user_stage.created_on = Utc::now();
         let _ = user_mgr
-            .add_stage(&plan.tenant, user_stage, &plan.create_option)
+            .add_stage(&plan.tenant, user_stage.clone(), &plan.create_option)
             .await?;
 
+        if user_stage.stage_type != StageType::External {
```

Comment:
We should algin with snowflake.
I guess snowflake doesn't remove the external stage files.
For internal stage it will create a different stage with the same name?

Reply:
create or replace equal:
drop stage ... - this doesn't mean remove any stage files.
create stage ...

Reply:
yes,  should not remove files in external stage

>> For internal stage it will create a different stage with the same name?

we do not have id for stage, so not able to do this.

Reply:
better if add a test for `not remove files in external stage`

Reply:
"drop stage ... - this doesn't mean remove any stage files."

tested snowflake, it will remove stage files after replace.